udata/core/topic/apiv2.py

```diff
@@ -127,37 +65,52 @@ def get(self, topic):
         """Get a given topic"""
         return topic
 
+    @apiv2.secure
+    @apiv2.doc("update_topic")
+    @apiv2.expect(topic_input_fields)
+    @apiv2.marshal_with(topic_fields)
+    @apiv2.response(400, "Validation error")
+    @apiv2.response(403, "Forbidden")
+    def put(self, topic):
+        """Update a given topic"""
+        if not TopicEditPermission(topic).can():
+            apiv2.abort(403, "Forbidden")
+        form = apiv2.validate(TopicForm, topic)
+        return form.save()
 
-topic_add_items_fields = apiv2.model(
-    "TopicItemsAdd",
-    {
-        "id": fields.String(description="Id of the item to add", required=True),
-    },
-    location="json",
-)
+    @apiv2.secure
+    @apiv2.doc("delete_topic")
+    @apiv2.response(204, "Object deleted")
+    @apiv2.response(403, "Forbidden")
+    def delete(self, topic):
+        """Delete a given topic"""
+        if not TopicEditPermission(topic).can():
+            apiv2.abort(403, "Forbidden")
+        # Remove discussions linked to the topic
+        Discussion.objects(subject=topic).delete()
+        topic.delete()
+        return "", 204
 
 
-@ns.route("/<topic:topic>/datasets/", endpoint="topic_datasets", doc=common_doc)
-class TopicDatasetsAPI(API):
-    @apiv2.doc("topic_datasets")
-    @apiv2.expect(dataset_parser.parser)
-    @apiv2.marshal_with(dataset_page_fields)
+@ns.route("/<topic:topic>/elements/", endpoint="topic_elements", doc=common_doc)
+class TopicElementsAPI(API):
+    @apiv2.doc("topic_elements")
+    @apiv2.expect(elements_parser.parser)
+    @apiv2.marshal_with(element_page_fields)
     def get(self, topic):
-        """Get a given topic datasets, with filters"""
-        args = dataset_parser.parse()
-        args["topic"] = topic.id
-        datasets = Dataset.objects(archived=None, deleted=None, private=False)
-        datasets = dataset_parser.parse_filters(datasets, args)
-        sort = args["sort"] or ("$text_score" if args["q"] else None) or "-created_at_internal"
-        return datasets.order_by(sort).paginate(args["page"], args["page_size"])
+        """Get a given topic's elements with pagination."""
+        args = elements_parser.parse()
+        elements = elements_parser.parse_filters(
+            TopicElement.objects(id__in=[e.pk for e in topic.elements]),
+            args,
+        )
+        return elements.paginate(args["page"], args["page_size"])
 
     @apiv2.secure
-    @apiv2.doc("topic_datasets_create")
-    @apiv2.expect([topic_add_items_fields])
+    @apiv2.doc("topic_elements_create")
+    @apiv2.expect([topic_add_elements_fields])
```

Comment:
I'm confused about this expect. `topic_add_elements_fields` is only composed of an `id` but we actually need a nested `element` with `class` + `id` infos?
Don't we want `[element_fields]`?